addons/web/static/src/scss/fields.scss

```diff
@@ -96,6 +96,8 @@
         }
 
         .badge {
+            display: flex;
+            max-width: 100%;
```

Comment:
you should also add a max-width:100% on the .o_field_many2manytags div i think (to prevent badges from overflowing in readonly)

Reply:
done